tests/test_types_typeddict.py

```diff
@@ -605,31 +604,6 @@ class RecursiveGenTypedDict(TypedDict, Generic[T]):
         ls: List[T]
 
     int_data: RecursiveGenTypedDict[int] = {'foo': {'foo': None, 'ls': [1]}, 'ls': [1]}
-    with pytest.raises(
-        PydanticUserError,
-        match=re.escape(
-            '`RecursiveGenTypedDictModel[int]` is not fully defined; you should define `RecursiveGenTypedDict`,'
-            ' then call `RecursiveGenTypedDictModel[int].model_rebuild()` before the first'
-            ' `RecursiveGenTypedDictModel[int]` instance is created.'
-        ),
-    ):
-        IntModel(rec=int_data).rec
-
-
-def test_recursive_generic_typeddict_in_function_rebuild_pass():
-    T = TypeVar('T')
-
-    class RecursiveGenTypedDictModel(BaseModel, Generic[T]):
-        rec: 'RecursiveGenTypedDict[T]'
-
-    IntModel = RecursiveGenTypedDictModel[int]
-
-    class RecursiveGenTypedDict(TypedDict, Generic[T]):
-        foo: Optional['RecursiveGenTypedDict[T]']
-        ls: List[T]
-
-    int_data: RecursiveGenTypedDict[int] = {'foo': {'foo': None, 'ls': [1]}, 'ls': [1]}
-    IntModel.model_rebuild()
```

Comment:
This was removed because it was misbehavior documented in this test, and was fixed by another tweak.